src/lib/features/segment/segment-controller.ts

```diff
@@ -347,25 +347,48 @@ export class SegmentsController extends Controller {
         req: IAuthRequest<{ id: number }>,
         res: Response<SegmentStrategiesSchema>,
     ): Promise<void> {
-        const { id } = req.params;
+        const id = Number(req.params.id);
```

Comment:
This was being passed as a string all the way through to the end, which made the query fail to find anything. However, we type it as `number` everywhere after this, so let's actually do the conversion.